packages/edit-post/src/components/preferences-modal/index.js

```diff
@@ -301,34 +325,57 @@ export default function PreferencesModal() {
 		);
 	} else {
 		modalContent = (
-			<Navigation
-				activeMenu={ activeMenu }
-				onActivateMenu={ setActiveMenu }
-			>
-				<NavigationMenu menu={ PREFERENCES_MENU }>
-					{ tabs.map( ( tab ) => {
-						return (
-							<NavigationItem
-								key={ tab.name }
-								title={ tab.title }
-								navigateToMenu={ tab.name }
-							/>
-						);
-					} ) }
-				</NavigationMenu>
-				{ sections.map( ( section ) => {
-					return (
-						<NavigationMenu
-							key={ `${ section.name }-menu` }
-							menu={ section.name }
-							title={ section.tabLabel }
-							parentMenu={ PREFERENCES_MENU }
-						>
-							<NavigationItem>{ section.content }</NavigationItem>
-						</NavigationMenu>
-					);
-				} ) }
-			</Navigation>
+			<Card isBorderless>
+				<CardBody>
+					<Navigator initialPath="/">
+						<NavigatorScreen path="/">
+							<ItemGroup>
+								{ tabs.map( ( tab ) => {
+									return (
+										<NavigationButton
+											key={ tab.name }
+											path={ tab.name }
+											as={ Item }
+											isAction
+										>
+											<HStack justify="space-between">
```

Comment:
Why we are using `Hstack` here? Couldn't we use `MenuItem` or just `Button` component? I'm catching up with the new Navigator component and try to understand the reasoning..

Reply:
Because I want a "chevron" icon at the right of the button

Reply:
`MenuItem` does just that by default and by testing it before, has no `spacing` issues described above. It aligns with the `x` icon. Also `MenuItem` and `Button` have `iconPosition as well.

Reply:
Interesting, I thought that component was only used in dropdowns. I can give it a try.

Reply:
cc @ciampo if you have any more info about that component and whether you agree that it's a good fit for this piece of UI.

Reply:
Maybe I'm wrong but it seems `ItemGroup` and `Item` kind of duplicate what `MenuGroup` and `MenuItem` are for. What's the story for these, should we merge them?

Reply:
I am not an expert of `MenuGroup` and `MenuItem`, so I'll try to answer to the best of my knowledge.

>  I thought that component was only used in dropdowns.

That was also my assumption.

Apart from that, it looks like `ItemGroup` and `Item` are more low-level (and more general) than `MenuGroup` and `MenuItem`.

`ItemGroup` and `Item` are very much about composability with other components, and are not "opinionated" about the contents/layout of each `Item` (which is why, for example, Riad used a `HStack` component) or about which element is used to render the `Item` (while `MenuItem` forces it to be a `Button`).


> should we merge them?

I don't think so — as I said before, `MenuGroup` and `MenuItem` seem to represent a more specific need than `ItemGroup` and `Item`. At most, we could look into refactoring `MenuGroup` and `MenuItem` to use `ItemGroup` and `Item` under the hood (although I'm not 100% sure, it would require some investigation)

Reply:
Also, apparently, they can't be used in any UI that needs to render in the native editor (see https://github.com/WordPress/gutenberg/pull/35191#issue-1009876310)

Reply:
Took a look at both approaches I think I have some issues with both:

 - Using MenuGroup and MenuItem: I felt that these had simpler APIs to achieve this kind of menus pretty quickly with no hassle. The spacing seemed indeed a bit better and match the other components (modal). That said, I think the semantics are broken because these are meant to be used inside `DropdownMenu` which is supposed to render the `role="menu"` and the group would add the `role="group"` and the items the `role="menuitem"`. 
 - Using ItemGroup and Item: These seem rather low level without any semantics, creating the desired result requires more work and tweaking and the spacing might not align properly with modals.

---
I think my ideal solution would be to have `MenuGroup` and `MenuItem` refactored to rely on the lower level `ItemGroup` and `Item`, have them support menus semantics outside dropdowns properly. (Maybe a new `Menu` component to serve as a wrapper for everything) and use consistent spacing between all the components.

What do you think @ciampo Should we get an issue at least with some of that?

For the moment, for this PR, I'll be leaving it as is. I think it's good enough to ship like that until we figure out how to merge/fix these components (ideally before we make all of them stable and confuse users).

Reply:
I opened #35210 and added it to the [`ItemGroup` tracking issue](https://github.com/WordPress/gutenberg/issues/34709)

> (Maybe a new Menu component to serve as a wrapper for everything)

Not sure I understand what you're suggesting here

Reply:
Right now when you you insert `MenuGroup` it inserts a div with `role="group"`, and when you insert `MenuItem` it inserts a button with `role="menuitem"`. I think these require a wrapper with a `role="menu"` to have the right semantics. In the case of dropdowns, that wrapper is added in `DropdownMenu` but maybe it should be its own component (for usage outside dropdowns)

Reply:
Thank you — I replied directly in the new issue